mzLib/UsefulProteomicsDatabases/ProteinDbWriter.cs

```diff
@@ -5,12 +5,283 @@
+                var allRelevantModifications =
+                    new HashSet<Modification>(nucleicAcidList
+                        .SelectMany(p => p.OneBasedPossibleLocalizedModifications.SelectMany(m => m.Value))
+                        .Concat(additionalModsToAddToProteins
+                            .Where(n => nucleicAcidList.Select(nu => nu.Accession).Contains(n.Key))
+                            .SelectMany(kv => kv.Value.Select(v => v.Item2))));
+
+                foreach (Modification mod in allRelevantModifications.OrderBy(m => m.IdWithMotif))
+                {
+                    writer.WriteStartElement("modification");
+                    writer.WriteString(mod.ToString() + Environment.NewLine + "//");
+                    writer.WriteEndElement();
+                }
+
+                foreach (var nucleicAcid in nucleicAcidList)
+                {
+                    writer.WriteStartElement("entry");
+                    writer.WriteStartElement("accession");
+                    writer.WriteString(nucleicAcid.Accession);
+                    writer.WriteEndElement();
+
+                    if (nucleicAcid.Name.IsNotNullOrEmptyOrWhiteSpace())
+                    {
+                        writer.WriteStartElement("name");
+                        writer.WriteString(nucleicAcid.Name);
+                        writer.WriteEndElement();
+                    }
+
+                    if (nucleicAcid.FullName.IsNotNullOrEmptyOrWhiteSpace())
+                    {
+                        writer.WriteStartElement("protein");
+                        writer.WriteStartElement("recommendedName");
+                        writer.WriteStartElement("fullName");
+                        writer.WriteString(nucleicAcid.FullName);
+                        writer.WriteEndElement();
+                        writer.WriteEndElement();
+                        writer.WriteEndElement();
+                    }
+
+                    writer.WriteStartElement("gene");
+                    foreach (var geneName in nucleicAcid.GeneNames)
+                    {
+                        writer.WriteStartElement("name");
+                        writer.WriteAttributeString("type", geneName.Item1);
+                        writer.WriteString(geneName.Item2);
+                        writer.WriteEndElement();
+                    }
+                    writer.WriteEndElement();
+
+                    if (nucleicAcid.Organism.IsNotNullOrEmptyOrWhiteSpace())
+                    {
+                        writer.WriteStartElement("organism");
+                        writer.WriteStartElement("name");
+                        writer.WriteAttributeString("type", "scientific");
+                        writer.WriteString(nucleicAcid.Organism);
+                        writer.WriteEndElement();
+                        writer.WriteEndElement();
+                    }
+
+                    //foreach (var dbRef in nucleicAcid)
```

Comment:
please add a "comment" explaining when all theses comments go away and how.

Reply:
added it to the remarks section of the method header